packages/neovim/src/utils/transport.ts

```diff
@@ -111,7 +114,19 @@ class Transport extends EventEmitter {
       iter.next().then(resolved => {
         if (!resolved.done) {
           if (!Array.isArray(resolved.value)) {
-            throw new TypeError('expected msgpack result to be array');
+            const resultType =
+              resolved.value == null
+                ? 'null or undefined'
+                : typeof resolved.value;
+            this.logger.debug(
```

Comment:
to be safe, let's not involve the logger. this is low-level code and we don't want to get into loops or anything like that. 

- instead of printing the type, let's print the actual `resolved.value` use [node.util.inspect](https://nodejs.org/api/util.html#utilinspectobject-options).
    - example: https://github.com/neovim/node-client/blob/b05c7e60522d7097b504321b63b1db46752c8f89/packages/neovim/src/utils/logger.ts#L2-L21
- to be extra safe, let's wrap the inspect() in a try-catch.

```typescript
let valstr = '?'
try {
  valstr = inspect(resolved.value, { sorted: true, maxArrayLength: 10, maxStringLength: 500, compact: true, breakLength = 500 })
} catch {}
```

Reply:
What do you think now? I put a console.error within the new catch block since I don't like the idea of swallowing the error.

Also do I need to add unit tests to pass CI for this change?

Reply:
> I put a console.error within the new catch block since I don't like the idea of swallowing the error.

That's usually a good instinct, but in this case the failure will be signaled because we'll show "?" in the thrown error. And this should be rare, so better to wait until it's a problem since this is a low-level codepath.